src/Common/DataObjects/Pitch.ts

```diff
@@ -347,6 +347,20 @@ export class Pitch {
         ", Note: " + this.fundamentalNote + ", octave: " + this.octave.toString();
     }
 
+    public IndexOf(array: Array<Pitch>, start: number = 0): number {
```

Comment:
I think this is not general enough - rather a special case and therefore not so good inside the Pitch class. Better move it to the StafflineNoteCalculator.